src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/WebSocketDeflateOptions.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Net.WebSockets
+{
+    /// <summary>
+    /// Options to enable per-message deflate compression for <seealso cref="WebSocket" />.
+    /// </summary>
+    public sealed class WebSocketDeflateOptions
+    {
+        private int _clientMaxWindowBits = 15;
+        private int _serverMaxWindowBits = 15;
+
+        /// <summary>
+        /// This parameter indicates the base-2 logarithm of the LZ77 sliding window size of the client context.
+        /// Must be a value between 9 and 15. The default is 15.
```

Comment:
Yes, but there is a problem. The underlying zlib component doesn't support 8 bits in deflater (see https://github.com/madler/zlib/issues/94#issuecomment-125832411 and https://zlib.net/manual.html). Quote from the manual "For the current implementation of deflate(), a windowBits value of 8 (a window size of 256 bytes) is not supported.". I initially thought that we can silently use 9 instead of 8 when 8 is requested, but although the deflate will succeed, the inflate on the other side will not. The reason is that the websocket produces raw deflate stream and thus it needs to know the window bits in advance and cannot infer them from any gzip headers. Also take a look at https://github.com/madler/zlib/issues/171. 

Since the whole world uses the zlib library by Mark Adler, I am doubtful that any server or client really supports 8 bits.

Reply:
Great, put that comment here in the code so someone doesn't try to 'fix' it later.